02.calendar/cal.rb

```diff
@@ -0,0 +1,86 @@
+#!/usr/bin/env ruby
+
+require 'date'
+require 'optparse'
+
+params = ARGV.getopts("y:m:")
+
+request_year = params["y"].to_i #入力した年
+request_month = params["m"].to_i #入力した月
+
+if request_year == 0 && request_month == 0
+  now = Date.today
+  this_year = now.year # 今年
+  this_month = now.month #今月
+  first_day = Date.new(now.year,now.month,1) #今月1日
+  last_day = Date.new(now.year,now.month,-1) #今月末日
+elsif request_year == 0 && request_month != 0
+  now = Date.today
+  this_year = now.year
+  first_day = Date.new(this_year,request_month,1)
+  last_day = Date.new(this_year,request_month,-1)
+else
+  first_day = Date.new(request_year,request_month,1)
+  last_day = Date.new(request_year,request_month,-1)
+end
```

Comment:
この処理は出力する月初と月末の Date インスタンスを変数に代入することですね
以下とかいかがでしょ

```suggestion
output_year = request_year.zero? ? Date.today.year : request_year
output_month = request_month.zero? ? Date.today.month : request_month
output_first_date = Date.new(output_year, output_month, 1)
output_last_date = Date.new(output_year, output_month, -1)
```

this_* という命名も良いですが、コード的により期待しているのは「出力対象の年と月」ですよね

また、 *_day という命名だと日付の数字なのか Date インスタンスなのかコードを追わないと分からないので、少し分かりやすいように *_date としました

Reply:
こちら有りがたく使わせていただきます!
書き方がとても勉強になりました!!